### DISCUSSION TITLE
Sort a yaml file by the top level node while excluding certain lines from the sorting

### DISCUSSION THREAD

**Discussion by thleo** [mikefarah/yq#1793] [Q&A]
**Please describe your feature request.**
I have a file with one top level parent that I want to remain at the top of the document, and then I have many other parents that I want to sort on a child key. I want to be able to sort only those subsequent parent nodes and leave the specific parent in whatever place in the file it exists in.

Note:
- how to questions should be posted in the discussion board and not raised as an issue.
- V3 will no longer have any enhancements.

**Describe the solution you'd like**
If we have data1.yml like:

```yaml
cantrip: Australia
  - animal: kangaroos
  - animal: bears
  - animal: aardvark
awsFunctionminimor:
 - name: minimor
awsFunctionfabulous:
 - name: fabulous
awsFunctiondancingpear:
 - name: dancingpear
```

And we run a command:

```bash
yq '. |= sort_by(.) && ignorelines=[(1,4)]' data1.yml
```

it could output

```yaml
cantrip: Australia
  - animal: kangaroos
  - animal: bears
  - animal: aardvark
awsFunctiondancingpear:
 - name: dancingpear
awsFunctionfabulous:
 - name: fabulous
 awsFunctionminimor:
 - name: minimor
```

**Describe alternatives you've considered**
I've tried running `yq '. |= sort_by(.)' data1.yml` which doesn't work as the node is not an array.

**Additional context**

**Comment by mikefarah** [comment]
TLDR:

```
yq 'sort_keys(.) | pick( (["cantrip"] + keys) | unique)' examples/data1.yaml
```

Explanation:
- First we sort the all keys of the map.
- You want 'cantrip' to go first, we can use `pick` to specify the map order. Pick takes an array of keys, and will pick those keys in the specified order.
- `["cantrip"] + keys` will have the order you want, but "cantrip" will be duplicated, as its also in `keys`
- we pipe that array through unique, which will drop duplicates while maintaining order
